include/SugarFolders/SugarFolders.php

```diff
@@ -1537,15 +1540,24 @@ protected function shouldFolderDisplay(array $folders, string $folderId): bool
         }
 
         foreach ($folders as $folder) {
+            // Check if unselected in user preferences
             $isSelected = $folder['selected'] ?? false;
             if (isFalse($isSelected)) {
                 continue;
             }
-            $id = $folder['id'] ?? '';
-
-            if ($id === $folderId) {
+            // Check parent ID
+            if ($folder['id'] === $folderId) {
```

Comment:
Hi @MikeyJC, why did we remove the null check here (`?? '';`) here?

Reply:
Made the suggested change back to include null check